CMakeLists.txt

```diff
@@ -366,6 +369,10 @@ if(JANSSON_BUILD_SHARED_LIBS)
       ${JANSSON_HDR_PUBLIC}
       src/jansson.def)
 
+   if (LIBC_IS_GLIBC)
+      set(CMAKE_SHARED_LINKER_FLAGS "${CMAKE_SHARED_LINKER_FLAGS} -Wl,--default-symver")
```

Comment:
In the unlikely event that jansson adds a second shared library, this use of `CMAKE_SHARED_LINKER_FLAGS` assumes that you'll want this linker flag for *all* shared libraries (which would probably be true, tbh). This avoids a dependency on a newer CMake, which would be required if I'd used `target_link_options()`.